tf_agents/environments/wrappers_test.py

```diff
@@ -914,5 +914,60 @@ def test_observation_and_action_stacked(self):
     self.assertEqual([5, 6, 7], time_step.observation['action'].tolist())
 
 
+class CustomMultiDiscretePyEnv(py_environment.PyEnvironment):
+  def __init__(self):
+    super(CustomMultiDiscretePyEnv, self).__init__()
+    self._position = np.array([0, 0], dtype=np.int32)
+
+  def _reset(self):
+    self._position = np.array([0, 0], dtype=np.int32)
+    return ts.restart(self._position.copy())
+
+  def observation_spec(self):
+    return array_spec.BoundedArraySpec(shape=(2, 2), dtype=np.int32, minimum=0, maximum=1)
+
+  def action_spec(self):
+    return array_spec.BoundedArraySpec(shape=(2, 2), dtype=np.int32, minimum=0, maximum=1)
+
+  def _step(self, action):
+    self._position += action
+    if self._position[0] + self._position[1] < 2:
+      return ts.transition(self._position.copy(), 1)
+    return ts.termination(self._position.copy(), 1)
+
+
+class MultiDiscreteToDiscreteWrapperTest(absltest.TestCase):
+
+  def test_new_action_spec_shape(self):
+    env = CustomMultiDiscretePyEnv()
+    wrapped_env = wrappers.MultiDiscreteToDiscreteWrapper(env)
+    self.assertEqual((4, ), wrapped_env.action_spec().shape)
+
+  def test_new_and_old_actions_have_one_to_one_correspondence(self):
+    env = CustomMultiDiscretePyEnv()
+    wrapped_env = wrappers.MultiDiscreteToDiscreteWrapper(CustomMultiDiscretePyEnv())
+    ts = env.step((0, 0))
+    wrapped_ts = wrapped_env.step(0)
+    self.assertEqual(ts.observation.tolist(), wrapped_ts.observation.tolist())
+
+    env = CustomMultiDiscretePyEnv()
+    wrapped_env = wrappers.MultiDiscreteToDiscreteWrapper(CustomMultiDiscretePyEnv())
+    ts = env.step((0, 1))
```

Comment:
Would this fail if you did wrappedenv.step(2) Make sure the test covers a case where the ordering of the mapping changes.

Maybe make the cusotmenvs position change by 10*action[0] + action[1]?